File: test_ws.js
const WebSocket = require('ws');
const apiKey = "dummy";
const url = 'wss://api.deepgram.com/v1/listen?model=nova-3&encoding=linear16&sample_rate=16000&channels=1';
const ws = new WebSocket(url, {
  headers: { Authorization: `Token ${apiKey}` },
});
ws.on('open', () => { console.log('open'); ws.close(); });
ws.on('error', (err) => { console.log('error', err.message); });
ws.on('close', (code, reason) => { console.log('close', code, reason.toString()); });

File: test_ws.py
import asyncio
import websockets
import os

async def test():
    key = os.environ.get("DEEPGRAM_API_KEY")
    url = "wss://api.deepgram.com/v1/listen?model=nova-3&encoding=linear16&sample_rate=48000&channels=1&detect_language=true"
    headers = {"Authorization": f"Token {key}"}
    
    try:
        async with websockets.connect(url, additional_headers=headers) as ws:
            print("Successfully connected!")
            ws.close()
    except Exception as e:
        print("Failed:", e)

asyncio.run(test())
